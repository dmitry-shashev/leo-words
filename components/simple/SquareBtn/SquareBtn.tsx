import { FC } from 'react'

interface Props {
  label: string
  onClick: () => void
  ariaLabel?: string
}

export const SquareBtn: FC<Props> = ({ label, onClick, ariaLabel }) => {
  return (
    <button
      onClick={onClick}
      aria-label={ariaLabel ?? label}
      className="w-10 h-10 text-gray-400 text-xl opacity-30"
      type="button"
    >
      {label}
    </button>
  )
}
