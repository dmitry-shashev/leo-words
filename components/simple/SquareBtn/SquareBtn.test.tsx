import { expect, test, vi } from 'vitest'
import { fireEvent, render } from '@testing-library/react'
import { SquareBtn } from '@/components/simple/SquareBtn/SquareBtn'

test('SquareBtn', () => {
  const onClick = vi.fn()

  const { asFragment, getByRole } = render(
    <SquareBtn onClick={onClick} label="Some sqr btn" />
  )

  expect(asFragment()).toMatchSnapshot()

  expect(onClick).toBeCalledTimes(0)
  fireEvent.click(getByRole('button'))
  expect(onClick).toBeCalledTimes(1)
})
