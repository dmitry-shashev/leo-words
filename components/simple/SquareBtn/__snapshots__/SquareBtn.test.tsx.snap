// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SquareBtn 1`] = `
<DocumentFragment>
  <button
    aria-label="Some sqr btn"
    class="w-10 h-10 text-gray-400 text-xl opacity-30"
    type="button"
  >
    Some sqr btn
  </button>
</DocumentFragment>
`;
